src/rpcmining.cpp

```diff
@@ -211,7 +211,7 @@ UniValue setgenerate(const UniValue& params, bool fHelp)
     if (params.size() > 0)
         fGenerate = params[0].get_bool();
 
-    int nGenProcLimit = -1;
+    int nGenProcLimit = GetArg("-genproclimit", DEFAULT_GENERATE_THREADS);
```

Comment:
Why this changes from -1 to ```GetArg("-genproclimit", DEFAULT_GENERATE_THREADS)``` but in ```getmininginfo()``` the change is from -1 to DEFAULT_GENERATE_THREADS?